Dockerfile

```diff
@@ -0,0 +1,213 @@
+    cd build ; \
+    cmake ../moab -DENABLE_HDF5=ON \
+                -DENABLE_NETCDF=ON \
+                -DBUILD_SHARED_LIBS=OFF \
+                -DENABLE_FORTRAN=OFF \
+                -DCMAKE_INSTALL_PREFIX=/MOAB ; \
+    make -j"$compile_cores" ; \
+    make -j"$compile_cores" install ; \
+    rm -rf * ; \
+    cmake ../moab -DBUILD_SHARED_LIBS=ON \
+                -DENABLE_HDF5=ON \
+                -DENABLE_PYMOAB=ON \
+                -DENABLE_BLASLAPACK=OFF \
+                -DENABLE_FORTRAN=OFF \
+                -DCMAKE_INSTALL_PREFIX=/MOAB ; \
+    make -j"$compile_cores" ; \
+    make -j"$compile_cores" install ; \
+    cd pymoab ; \
+    bash install.sh ; \
+    python setup.py install ; \
+    fi
+
+
+# Clone and install Double-Down
+RUN if [ "$include_neutronics" = "true" ] ; \
+    then git clone https://github.com/pshriwise/double-down ; \
+    cd double-down ; \
+    mkdir build ; \
+    cd build ; \
+    cmake .. -DCMAKE_INSTALL_PREFIX=.. \
+        -DMOAB_DIR=/MOAB \
+        -DEMBREE_DIR=/embree/lib/cmake/embree-3.12.1 \
+        -DEMBREE_ROOT=/embree/lib/cmake/embree-3.12.1 ; \
+    make -j"$compile_cores" ; \
+    make -j"$compile_cores" install ; \
+    fi
+
+# Clone and install DAGMC
+RUN if [ "$include_neutronics" = "true" ] ; \
+    then mkdir DAGMC ; \
+    cd DAGMC ; \
+    git clone -b develop https://github.com/svalinn/dagmc ; \
+    mkdir build ; \
+    cd build ; \
+    cmake ../dagmc -DBUILD_TALLY=ON \
+        -DCMAKE_INSTALL_PREFIX=/dagmc/ \
+        -DMOAB_DIR=/MOAB \
+        -DBUILD_STATIC_LIBS=OFF \
+        -DBUILD_STATIC_EXE=OFF ; \
+    make -j"$compile_cores" install ; \
+    rm -rf /DAGMC/dagmc /DAGMC/build ; \
+    fi
+
+# Clone and install OpenMC with DAGMC
+RUN if [ "$include_neutronics" = "true" ] ; \
+    then git clone --recurse-submodules https://github.com/openmc-dev/openmc.git /opt/openmc ; \
+    cd /opt/openmc ; \
+    mkdir build ; \
+    cd build ; \
+    cmake -Doptimize=on -Ddagmc=ON \
+        -DDAGMC_DIR=/DAGMC/ \
+        -DHDF5_PREFER_PARALLEL=on ..  ; \
+    make -j"$compile_cores" ; \
+    make -j"$compile_cores" install ; \
+    cd ..  ; \
+    pip install -e .[test] ; \
+    /opt/openmc/tools/ci/download-xs.sh ; \
+    fi
+
+ENV OPENMC_CROSS_SECTIONS=/root/nndc_hdf5/cross_sections.xml
+
+# Copies over the Paramak code from the local repository
+
+RUN if [ "$include_neutronics" = "true" ] ; \
+    then pip install neutronics_material_maker ; \
+    pip install parametric_plasma_source ; \
+    fi
+
+COPY requirements.txt requirements.txt
+RUN pip install -r requirements.txt
+
+
+# Copy over the source code, examples and tests
+COPY run_tests.sh run_tests.sh
+COPY paramak paramak/
+COPY examples examples/
+COPY setup.py setup.py
+COPY tests tests/
+COPY README.md README.md
+
+# using setup.py instead of pip due to https://github.com/pypa/pip/issues/5816
+RUN python setup.py install
+
+WORKDIR examples
```

Comment:
**BUG RISK**
I was attempting to run one of the examples inside of the docker container and hit this error:

```
(base) root@07c6f4d85ce9:/# python examples/example_neutronics_simulations/ball_reactor.py 
/opt/conda/lib/python3.7/site-packages/neutronics_material_maker/utils.py:14: UserWarning: OpenMC not found, .openmc_material, .serpent_material, .mcnp_material,            .fispact_material not avaiable
  .fispact_material not avaiable")
/opt/conda/lib/python3.7/site-packages/neutronics_material_maker/material.py:29: UserWarning: OpenMC python package not found, .openmc_material, .serpent_material,             .mcnp_material, .fispact_material methods not avaiable
  .mcnp_material, .fispact_material methods not avaiable")
/opt/conda/lib/python3.7/site-packages/neutronics_material_maker/mutlimaterial.py:21: UserWarning: OpenMC python package not found, .openmc_material, .serpent_material,             .mcnp_material, .fispact_material methods not avaiable
  .mcnp_material, .fispact_material methods not avaiable")
/opt/conda/lib/python3.7/site-packages/paramak-0.1.0-py3.7.egg/paramak/parametric_neutronics/neutronics_model_from_reactor.py:14: UserWarning: parametric_plasma_source not found distributed plasma             sources are not avaialbe in Neutronics simulations
  sources are not avaialbe in Neutronics simulations', UserWarning)
/opt/conda/lib/python3.7/site-packages/paramak-0.1.0-py3.7.egg/paramak/parametric_neutronics/neutronics_model_from_reactor.py:20: UserWarning: OpenMC not found, NeutronicsModelFromReactor.simulate             method not available
  method not available', UserWarning)
Traceback (most recent call last):
  File "examples/example_neutronics_simulations/ball_reactor.py", line 66, in <module>
    make_model_and_simulate()
  File "examples/example_neutronics_simulations/ball_reactor.py", line 39, in make_model_and_simulate
    temperature_in_K=500),
  File "/opt/conda/lib/python3.7/site-packages/neutronics_material_maker/material.py", line 206, in __init__
    self._populate_from_inbuilt_dictionary()
  File "/opt/conda/lib/python3.7/site-packages/neutronics_material_maker/material.py", line 745, in _populate_from_inbuilt_dictionary
    "enrichment_target"
  File "/opt/conda/lib/python3.7/site-packages/neutronics_material_maker/material.py", line 528, in enrichment_target
    if value not in openmc.data.NATURAL_ABUNDANCE.keys():
NameError: name 'openmc' is not defined
```
It would appear that one of the projects that paramak depends on (openmc) isn't installed in the Dockerfile or it's not available in the default environment.

--------------

When we were able to run the tests after fixing the issue above , we came across a string of warnings:

```
==================================== warnings summary =====================================
paramak/shape.py:5
  /paramak/shape.py:5: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
    from collections import Iterable

paramak/parametric_neutronics/neutronics_model_from_reactor.py:14
  /paramak/parametric_neutronics/neutronics_model_from_reactor.py:14: UserWarning: parametric_plasma_source not found distributed plasma             sources are not avaialbe in Neutronics simulations
    sources are not avaialbe in Neutronics simulations', UserWarning)

paramak/parametric_neutronics/neutronics_model_from_reactor.py:20
  /paramak/parametric_neutronics/neutronics_model_from_reactor.py:20: UserWarning: OpenMC not found, NeutronicsModelFromReactor.simulate             method not available
    method not available', UserWarning)

opt/conda/lib/python3.7/site-packages/neutronics_material_maker/utils.py:14
  /opt/conda/lib/python3.7/site-packages/neutronics_material_maker/utils.py:14: UserWarning: OpenMC not found, .openmc_material, .serpent_material, .mcnp_material,            .fispact_material not avaiable
    .fispact_material not avaiable")

opt/conda/lib/python3.7/site-packages/neutronics_material_maker/material.py:29
  /opt/conda/lib/python3.7/site-packages/neutronics_material_maker/material.py:29: UserWarning: OpenMC python package not found, .openmc_material, .serpent_material,             .mcnp_material, .fispact_material methods not avaiable
    .mcnp_material, .fispact_material methods not avaiable")

opt/conda/lib/python3.7/site-packages/neutronics_material_maker/mutlimaterial.py:21
  /opt/conda/lib/python3.7/site-packages/neutronics_material_maker/mutlimaterial.py:21: UserWarning: OpenMC python package not found, .openmc_material, .serpent_material,             .mcnp_material, .fispact_material methods not avaiable
    .mcnp_material, .fispact_material methods not avaiable")

...

tests/test_example_reactors.py::test_object_properties::test_make_parametric_ball_rector
tests/test_example_reactors.py::test_object_properties::test_make_parametric_single_null_ball_reactor
tests/test_parametric_reactors/test_BallReactor.py::test_BallReactor::test_creation_with_narrow_divertor
tests/test_parametric_reactors/test_BallReactor.py::test_BallReactor::test_divertor_upper_lower
tests/test_parametric_reactors/test_BallReactor.py::test_BallReactor::test_svg_creation
tests/test_parametric_reactors/test_BallReactor.py::test_BallReactor::test_with_pf_and_tf_coils
tests/test_parametric_reactors/test_BallReactor.py::test_BallReactor::test_with_pf_coils
tests/test_parametric_reactors/test_SegmentedBlanketBallReactor.py::test_SegmentedBlanketBallReactor::test_gap_between_blankets_impacts_volume
tests/test_parametric_reactors/test_SegmentedBlanketBallReactor.py::test_SegmentedBlanketBallReactor::test_number_of_blanket_segments_impacts_volume
  /paramak/parametric_components/blanket_fp.py:227: UserWarning:
  
  BlanketFP: Some points with negative R coordinate have been ignored.

tests/test_parametric_components/test_PoloidallySegmentedBlanketFP.py::test_BlanketFP::test_segment_angles_affects_solid
  /paramak/parametric_components/blanket_poloidal_segment.py:61: UserWarning:
  
  start_angle and stop_angle attributes will be ignored if segments_angles is not None

tests/test_parametric_components/test_casing.py::test_TFCoilCasing::test_creation
  /paramak/parametric_components/tf_coil_casing.py:31: UserWarning:
  
  Casing azimuth_placement_angle should be the same value as TFCoilCasing.magnet.

tests/test_parametric_components/test_casing.py::test_TFCoilCasing::test_creation
  /paramak/utils.py:226: RuntimeWarning:
  
  divide by zero encountered in true_divide

-- Docs: https://docs.pytest.org/en/stable/warnings.html
====================== 362 passed, 228 warnings in 402.34s (0:06:42) ======================
```
_There were more than those listed above, shortening for brevity._

It's best practice to ensure minimal warnings when running tests. In this case, there seem to be 228 warnings.

<p align="right">
    <a href="https://www.pullrequest.com/">
        <img width="15%" src="https://static.pullrequest.com/images/dashboard/pullrequest-logo.svg">
    </a>
</p>